src/DependencyInjection/Compiler/BusBuilder/BusBuilder.php

```diff
@@ -123,4 +125,24 @@ private function registerHandlerServiceLocator(ContainerBuilder $container, arra
 
         return $handlerId;
     }
+
+    /**
+     * @param ContainerBuilder $container
+     * @param array $commandsToAccept
+     */
+    private function updateDebugMappings(ContainerBuilder $container, array $commandsToAccept)
+    {
+        $mappings = [];
+
+        if ($container->hasParameter('tactician.debug.mappings')) {
+            $mappings = $container->getParameter('tactician.debug.mappings');
+        }
+
+        $mappings[$this->id()] = $commandsToAccept;
+
+        $container->setParameter(
+            'tactician.debug.mappings',
+            $mappings
+        );
```

Comment:
Actually we don't need to expose internals in a parameter in fact. 
I suggest to get rid of the RoutingDebugReport class and just inject mappings as an array in the command using `$container->getDefinition('tactician.command.debug_mapping')->replaceArgument(0, $mappings);`, building `$mappings` using the return value of [`$routing->commandToServiceMapping()`](https://github.com/thephpleague/tactician-bundle/blob/46326e8372cf9159d0e23b6848791e6700a48325/src/DependencyInjection/Compiler/CommandHandlerPass.php#L31) .

Reply:
Great idea, `RoutingDebugReport` became just an array wrapper without any additional logic.

I removed it, but ther is another container-related issue.
Looks like it's compiler passes order related.

I have following service definition in configuration:

```
tactician.command.debug_mapping:
        class: League\Tactician\Bundle\Command\DebugMappingCommand
        tags:
            - { name: console.command }
```

I couldn't add an argument inside compiler pass because servic doesn't exist at this point: `You have requested a non-existent service "taectician.command.debug_mapping"`.

So I wanted to set full definition inside compiler pass, in this case command is not registered - because of compiler passes order?

Now it's defined in configuration and compiler pass overrides it setting new definition - looks hacky, probably I misconfigured something.

Reply:
@chalasr this one is important, github hid it :(

Reply:
Right, I missed it! It fails because the testing container of CommandHandlerPassTest does not load the `services.yml` file, that is a good thing in fact.
In order to not couple the compiler pass to the bundle extension, I propose to wire the command only if the definition exists. Here is a patch including a test case for when the definition exists:

```diff
diff --git a/src/DependencyInjection/Compiler/CommandHandlerPass.php b/src/DependencyInjection/Compiler/CommandHandlerPass.php
index 4dd92c2..7a2cc10 100644
--- a/src/DependencyInjection/Compiler/CommandHandlerPass.php
+++ b/src/DependencyInjection/Compiler/CommandHandlerPass.php
@@ -2,12 +2,10 @@
 
 namespace League\Tactician\Bundle\DependencyInjection\Compiler;
 
-use League\Tactician\Bundle\Command\DebugMappingCommand;
 use League\Tactician\Bundle\DependencyInjection\Compiler\BusBuilder\BusBuildersFromConfig;
 use League\Tactician\Bundle\DependencyInjection\HandlerMapping\HandlerMapping;
 use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
 use Symfony\Component\DependencyInjection\ContainerBuilder;
-use Symfony\Component\DependencyInjection\Definition;
 
 /**
  * This compiler pass maps Handler DI tags to specific commands.
@@ -44,13 +42,9 @@ class CommandHandlerPass implements CompilerPassInterface
         $container->setAlias('tactician.handler.locator.symfony', $builders->defaultBus()->locatorServiceId());
         $container->setAlias('tactician.middleware.command_handler', $builders->defaultBus()->commandHandlerMiddlewareId());
 
-        // Setup debug command
-        $container->setDefinition(
-            'tactician.command.debug_mapping',
-            new Definition(
-                DebugMappingCommand::class,
-                [$mappings]
-            )
-        );
+        // Wire debug command
+        if ($container->hasDefinition('tactician.command.debug_mapping')) {
+            $container->getDefinition('tactician.command.debug_mapping')->addArgument($mappings);
+        }
     }
 }
diff --git a/tests/DependencyInjection/Compiler/CommandHandlerPassTest.php b/tests/DependencyInjection/Compiler/CommandHandlerPassTest.php
index aa77c5a..29e8771 100644
--- a/tests/DependencyInjection/Compiler/CommandHandlerPassTest.php
+++ b/tests/DependencyInjection/Compiler/CommandHandlerPassTest.php
@@ -2,6 +2,7 @@
 
 namespace League\Tactician\Bundle\Tests\DependencyInjection\Compiler;
 
+use League\Tactician\Bundle\Command\DebugMappingCommand;
 use League\Tactician\Bundle\DependencyInjection\Compiler\CommandHandlerPass;
 use League\Tactician\Bundle\DependencyInjection\HandlerMapping\ClassNameMapping;
 use League\Tactician\Bundle\DependencyInjection\HandlerMapping\HandlerMapping;
@@ -119,6 +120,41 @@ class CommandHandlerPassTest extends TestCase
         );
     }
 
+    public function test_handler_wires_debug_command()
+    {
+        $container = $this->containerWithConfig(
+            [
+                'default_bus' => 'bus.a',
+                'commandbus' => [
+                    'bus.a' => ['middleware' => []],
+                    'bus.b' => ['middleware' => []]
+                ]
+            ]
+        );
+
+        $routing = new Routing(['bus.a', 'bus.b']);
+        $routing->routeToBus('bus.a', FakeCommand::class, 'some.handler.a');
+        $routing->routeToBus('bus.b', FakeCommand::class, 'some.handler.b');
+        $routing->routeToAllBuses(OtherFakeCommand::class, 'some.other.handler');
+
+        $mapping = $this->prophesize(HandlerMapping::class);
+        $mapping->build($container, Argument::type(Routing::class))->willReturn($routing);
+
+        $container->register('tactician.command.debug_mapping', DebugMappingCommand::class);
+
+        (new CommandHandlerPass($mapping->reveal()))->process($container);
+
+        $this->assertSame(
+            [
+                [
+                   'bus.a' => $routing->commandToServiceMapping('bus.a'),
+                   'bus.b' => $routing->commandToServiceMapping('bus.b'),
+                ],
+            ],
+            $container->getDefinition('tactician.command.debug_mapping')->getArguments()
+        );
+    }
+
     private function containerWithConfig($config)
     {
         $container = new ContainerBuilder();

```

Reply:
> It fails because the testing container of CommandHandlerPassTest does not load the services.yml file
Ok, that explains :D

> In order to not couple the compiler pass to the bundle extension
:+1: 

> Here is a patch including a test case for when the definition exists
:heart: 

Thanks again!